### PR TITLE
help: print cli/parser help message if used

### PR DESCRIPTION
**Pull request by dawidd6**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

External commands using `cli/parser` currently can't produce a valid help message, because `brew` looks for `#:` comments on top of the command file. With this patch, it becomes possible to get the help message normally.

Discussion on the issue here: https://github.com/Homebrew/homebrew-linux-dev/pull/137

### CODE REVIEW

**Comment by issyl0** [comment]
Thanks, Dawid! This works for me. Here's the before/after:

```
$ cd $(brew --repo homebrew/linux-dev)
$ hub pr checkout 137
$ brew build-bottle-pr --help
  Warning: No help text in: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-linux-dev/cmd/brew-build-bottle-pr.rb
  Example usage:
    brew search [TEXT|/REGEX/]
  [...]

$ cd $(brew --repo)
$ hub pr checkout 6965
$ brew build-bottle-pr --help
  Usage: brew build-bottle-pr [--remote=user] [--dry-run] [--verbose] [--tap-dir] [--force] Submit a pull request to build a bottle for a formula.
  [...]
```

File: Library/Homebrew/help.rb [hunk 1]
```diff
@@ -72,6 +72,9 @@ def help(cmd = nil, flags = {})
       # Resume execution in `brew.rb` for unknown commands.
       return if path.nil?
 
+      # Resume execution in `brew.rb` for external commands using "cli/parser".
+      return if !flags[:internal_cmd] && path.read.match?("require .cli/parser.")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/help.rb]: Would love to figure out something else beyond reading the file here. Maybe require the file and look for the relevant method name (which could/should be part of `cli/parser` itself if it isn't already)?

Comment (dawidd6) [reply Library/Homebrew/help.rb]: Something like that:

```ruby
require path
methods.include?(:usage_banner)
```

?

Comment (MikeMcQuaid) [reply Library/Homebrew/help.rb]: Yeh, `:announce_args` or similar. Would be good to share logic with https://github.com/Homebrew/brew/blob/e47a6ac4cb8810ab687096f224e1fc4c0f0f980e/Library/Homebrew/dev-cmd/man.rb#L156-L166

Comment (dawidd6) [reply Library/Homebrew/help.rb]: Okay, gonna need more help with that :smile:

**Comment by MikeMcQuaid** [comment]
Thanks @dawidd6!

**Pull request merged**